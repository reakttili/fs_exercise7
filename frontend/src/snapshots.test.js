import React from 'react'
import SimpleBlog from './components/SimpleBlog'
import { Blog } from './components/Blog'
import renderer from 'react-test-renderer'

//https://benmccormick.org/2016/09/19/testing-with-jest-snapshots-first-impressions/
//https://hackernoon.com/how-to-snapshot-test-everything-in-your-redux-app-with-jest-fde305ebedea
describe('snap', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<SimpleBlog blog="blog" onClick="onclick" >Blog</SimpleBlog>).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders correctly', () => {

    const blog = {
      title: 'title',
      author: 'VB',
      url: 'www',
      likes: 5,
      adder: 'user',
      blogid: 100
    }
    const mockHandler = jest.fn()
    const ownProps = {
      bShowAll: true,
      blog:blog,
      parentRender:mockHandler
    }
    const tree = renderer.create(<Blog ownProps={ownProps} onClick="onclick" >Blog</Blog>).toJSON()
    expect(tree).toMatchSnapshot()
  })

})