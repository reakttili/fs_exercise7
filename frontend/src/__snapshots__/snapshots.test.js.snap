// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snap renders correctly 1`] = `
<div>
  <div
    className="titleauthordiv"
  >
     
  </div>
  <div
    className="likesdiv"
  >
    blog has 
     likes
    <button
      onClick="onclick"
    >
      like
    </button>
  </div>
</div>
`;

exports[`snap renders correctly 2`] = `
<div
  className="maindiv"
  style={
    Object {
      "border": "",
      "borderWidth": 0,
      "marginBottom": 20,
      "paddingLeft": 0,
      "paddingTop": 0,
    }
  }
>
  <div
    className="ui grid"
  >
    <div
      className="thirteen wide column"
    >
      <div>
        <table
          className="ui table"
        >
          <tbody
            className=""
          >
            <tr
              className=""
            >
              <td
                className=""
              >
                <div
                  className="namediv"
                  onClick={[Function]}
                >
                  <h1>
                    title
                     
                    VB
                  </h1>
                </div>
              </td>
            </tr>
            <tr
              className=""
            >
              <td
                className=""
              >
                5
                 likes
                <button
                  className="ui right floated button"
                  name="like"
                  onClick={[Function]}
                  role="button"
                >
                  like
                </button>
              </td>
            </tr>
            <tr
              className=""
            >
              <td
                className=""
              >
                <div>
                  <a
                    href="www"
                  >
                    www
                  </a>
                </div>
              </td>
            </tr>
            <tr
              className=""
            >
              <td
                className=""
              >
                <div>
                  added by: 
                  
                </div>
              </td>
            </tr>
          </tbody>
        </table>
        <h1>
          Comments
        </h1>
        <input
          name="commentInput"
        />
        <button
          name="addCommentButton"
          onClick={[Function]}
        >
          add comment
        </button>
      </div>
    </div>
    <div
      className="three wide column"
    />
  </div>
</div>
`;
